include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy_impl.h

```diff
@@ -66,7 +66,20 @@ namespace experimental{
 
     template<typename ...Args>
     selection_handle_t select(Args&&...) {
-      auto i = next_context_++ % num_contexts_;
+      size_t i=0;
+      universe_container_size_t current_context_;
+      while(true){
+          if(current_context_ == std::numeric_limits<universe_container_size_t>::max()){
+            universe_container_size_t new_context_ = (current_context_%num_contexts_)+1;
+            if(next_context_.compare_exchange_weak(current_context_, new_context_)){
+                i = new_context_;
+                break;
+            }
+          }else{
+              i = next_context_.fetch_add(1)%num_contexts_;
```

Comment:
still need to do compare_exchange in the else to be sure that value has not changed since the if condition was checked. Otherwise, by the time you reach this line it could now be max and so the wrap-around is necessary.

Reply:
Fixed.